File: app/src/main/java/com/example/plantonic/repo/LoginRepository.java
package com.example.plantonic.repo;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.plantonic.firebaseClasses.UserItem;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;


public class LoginRepository {

    private final FirebaseFirestore db = FirebaseFirestore.getInstance();

    private MutableLiveData<Boolean> _userExists = new MutableLiveData();
    public LiveData<Boolean> userExists = _userExists;

    public void checkIfUserExists(String userId){
        db.collection("users")
                .document(userId).get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
                    @Override
                    public void onComplete(@NonNull Task<DocumentSnapshot> task) {
                        _userExists.postValue(task.getResult().exists());
                    }
                });
    }

    public LiveData<Boolean> registerNewUser(UserItem userItem){
        db.collection("users")
                .document(userItem.getUserId())
                .set(userItem).addOnCompleteListener(new OnCompleteListener<Void>() {
                    @Override
                    public void onComplete(@NonNull Task<Void> task) {
                        _userExists.postValue(true);
                    }
                });
        return userExists;
    }
}

File: app/src/main/java/com/example/plantonic/repo/LoginRepository.kt
package com.example.plantonic.repo

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData
import com.example.plantonic.firebaseClasses.UserItem
import com.example.plantonic.retrofit.models.auth.AuthRequestModel
import com.example.plantonic.retrofit.models.auth.AuthResponseModel
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response


class LoginRepository : BaseRepository() {
    private val _userExists: MutableLiveData<Boolean?> = MutableLiveData()
    val userExists: LiveData<Boolean?> = _userExists

    private val _userToken: MutableLiveData<String?> = MutableLiveData()
    var userToken: LiveData<String?> = _userToken


    fun checkIfUserExists(userId: String) {
        db.collection("users")
            .document(userId).get()
            .addOnCompleteListener { task -> _userExists.postValue(task.result.exists()) }
    }

    fun registerNewUser(userItem: UserItem): LiveData<Boolean?> {
        db.collection("users")
            .document(userItem.userId)
            .set(userItem).addOnCompleteListener { _userExists.postValue(true) }
        return userExists
    }

    fun getJwtToken(userId: String?): LiveData<String?> {
        val uid = aes.encrypt(userId)
        Log.d("UID: ", uid)
        api.getUserToken(AuthRequestModel(uid))
            .enqueue(object : Callback<AuthResponseModel?> {
                override fun onResponse(
                    call: Call<AuthResponseModel?>,
                    response: Response<AuthResponseModel?>
                ) {
                    if (response.isSuccessful) {
                        val loginResponse = response.body()
                        if (loginResponse != null) {
                            _userToken.postValue(loginResponse.access_token)
                            Log.d("Login: ", "success")
                        }
                    } else {
                        _userToken.postValue(null)
                        Log.d("Login: ", "failed ${response.message()}")
                    }
                }

                override fun onFailure(call: Call<AuthResponseModel?>, t: Throwable) {
                    _userToken.postValue(null)
                    Log.d("Login: ", "failed" + t.message)
                    t.stackTrace
                }
            })

        return userToken
    }
}